pkg/build/registry/buildconfiginstantiate/rest.go

```diff
@@ -226,9 +243,32 @@ func (h *binaryInstantiateHandler) handle(r io.Reader) (runtime.Object, error) {
 	if err := exec.Stream(streamOptions); err != nil {
 		return nil, errors.NewInternalError(err)
 	}
+	cancel = false
 	return latest, nil
 }
 
+// cancelBuild will mark a build for cancellation unless
+// cancel is false in which case it is a no-op.
+func (h *binaryInstantiateHandler) cancelBuild(cancel bool, build *buildapi.Build) {
+	if !cancel {
```

Comment:
you're the one who wanted the defer logic wrapped into a func in the first place, now you're having me favor the func logic back out of the func....